lib/driver.js

```diff
@@ -1207,19 +1221,23 @@ class XCUITestDriver extends BaseDriver {
           return {device, realDevice: false, udid: device.udid};
         }
       } else {
-        // make sure it is a connected device. If not, the udid passed in is invalid
-        const devices = await getConnectedDevices();
-        this.log.debug(`Available devices: ${devices.join(', ')}`);
-        if (!devices.includes(this.opts.udid)) {
-          // check for a particular simulator
-          this.log.debug(`No real device with udid '${this.opts.udid}'. Looking for a simulator`);
-          try {
-            const device = await getSimulator(this.opts.udid, {
-              devicesSetPath: this.opts.simulatorDevicesSetPath,
-            });
-            return {device, realDevice: false, udid: this.opts.udid};
-          } catch (ign) {
-            throw new Error(`Unknown device or simulator UDID: '${this.opts.udid}'`);
+        if (this.skipDeviceCheck()) {
+          this.log.info(`Skipping connected device check. Using the given udid ${this.opts.udid}.`);
+        } else {
+          // make sure it is a connected device. If not, the udid passed in is invalid
+          const devices = await getConnectedDevices();
```

Comment:
not for this pr, but I just wondered if we could skip here with `appium:webDriverAgentUrl` since basically the caller may prepare device properly. So not only for this case.

Reply:
makes sense to me